src/service/dashboards/mod.rs

```diff
@@ -205,10 +202,10 @@ pub async fn move_dashboard(
     dashboard_id: &str,
     from_folder: &str,
     to_folder: &str,
-) -> Result<HttpResponse, io::Error> {
+) -> Result<HttpResponse, anyhow::Error> {
     if let Ok(dashboard) = db::dashboards::get(org_id, dashboard_id, from_folder).await {
         // make sure the destination folder exists
-        if db::folders::get(org_id, to_folder).await.is_err() {
+        if !db::folders::exists(org_id, to_folder).await? {
```

Comment:
I have the same comment here as in a previous comment:

> I think the previous logic may have had a subtle issue in that is assumed an `Err` response from `db::folders::get` indicates that the folder does not exist. This isn't strictly true since that function can return `Err` for a variety of reasons (eg: losing connection to the DB for some reason). Therefore I think that using `?` to fast-return an error from the call to the db function  should make this logic behave more correctly when the call to the db function happens to fail.

> I also introduced the `exists` function to make it more clear that the Boolean return value, rather then the `Ok`/`Err` status of the response itself is what communicates the existence of the folder.